### PR TITLE
UX Multichain: Added avatar group component

### PR DESCRIPTION
**Pull request by NidhiKJha**
## **Description**
This PR is to add a avatar group component required for the [avatars in account-list-item](https://github.com/MetaMask/MetaMask-planning/issues/1425)

## **Manual testing steps**

_1. Step1:_ Go to Storybook build
_2. Step2:_ Look for AvatarGroup Component
_3. Step3:_ Check if components are in a stacked manner
_4. Step4:_ Change the limit and check extra appear in a text like +2 format

## **Screenshots/Recordings**
Screenshot 2023-10-12 at 4 29 52 PM

## **Related issues**

_Fixes #21323 

## **Pre-merge author checklist**

- [x] I’ve followed [MetaMask Coding Standards](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/CODING_GUIDELINES.md).
- [x] I've clearly explained:
  - [x] What problem this PR is solving.
  - [x] How this problem was solved.
  - [x] How reviewers can test my changes.
- [x] I’ve indicated what issue this PR is linked to: Fixes #???
- [x] I’ve included tests if applicable.
- [x] I’ve documented any added code.
- [x] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/LABELING_GUIDELINES.md)).
- [x] I’ve properly set the pull request status:
  - [x] In case it's not yet "ready for review", I've set it to "draft".
  - [ ] In case it's "ready for review", I've changed it from "draft" to "non-draft".

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

File: ui/components/multichain/avatar-group/avatar-group.tsx [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+              name={x.symbol}
+              size={size}
+              borderColor={borderColor}
+            />
+          </Box>
+        ))}
+      </Box>
+      {showTag ? (
+        <Box>
+          {typeof tagValue === 'string' ? (
```
Comment (darkwing) [inline ui/components/multichain/avatar-group/avatar-group.tsx:65]: This is always going to be a `string`, right?

**Review by vthomas13 — DISMISSED** [review]

File: ui/components/multichain/avatar-group/avatar-group.tsx [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+  const tagValue = `+${(membersCount - limit).toLocaleString()}`;
+  return (
+    <Box
+      alignItems={AlignItems.center}
+      display={Display.Flex}
+      className={classnames('multichain-avatar-group', className)}
+      data-testid="avatar-group"
+    >
+      <Box display={Display.Flex}>
+        {visibleMembers.map((x, i) => (
```
Comment (darkwing) [inline ui/components/multichain/avatar-group/avatar-group.tsx:46]: Can we use a more descriptive name than `x`?  Maybe `token`?

Comment (darkwing) [reply ui/components/multichain/avatar-group/avatar-group.tsx:46]: Or `member`?

File: ui/components/multichain/avatar-group/avatar-group.tsx [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+          <Box
+            borderRadius={BorderRadius.full}
+            key={x.symbol}
+            style={
+              i === 0 ? { marginLeft: '0px' } : { marginLeft: marginLeftValue }
```
Comment (darkwing) [inline ui/components/multichain/avatar-group/avatar-group.tsx:51]: When the value is `0px` you can just put `0`

**Review by darkwing — CHANGES_REQUESTED** [review]
Minor nits, but this looks great!

File: ui/components/multichain/avatar-group/avatar-group.tsx [hunk 4]
```diff
@@ -35,6 +35,7 @@ export const AvatarGroup: React.FC<AvatarGroupProps> = ({
     marginLeftValue = '-16px';
   }
   const tagValue = `+${(membersCount - limit).toLocaleString()}`;
+  console.log(tagValue, typeof tagValue, showTag);
```
Comment (NidhiKJha) [inline ui/components/multichain/avatar-group/avatar-group.tsx]: This is removed here https://github.com/MetaMask/metamask-extension/pull/21342/commits/454b49938fdc7bdafe0f30f255604ba81c59d4c3

File: .vscode/settings.json [hunk 5]
```diff
@@ -2,5 +2,8 @@
   "editor.defaultFormatter": "rvest.vs-code-prettier-eslint",
   "editor.tabSize": 2,
   "files.trimTrailingWhitespace": true,
-  "javascript.preferences.importModuleSpecifier": "relative"
+  "javascript.preferences.importModuleSpecifier": "relative",
+  "[typescriptreact]": {
+    "editor.defaultFormatter": "esbenp.prettier-vscode"
+  }
```
Comment (NidhiKJha) [inline .vscode/settings.json]: Removed here https://github.com/MetaMask/metamask-extension/pull/21342/commits/9582342c3801b1e22675959d6dccd7388e598e8e

File: ui/components/multichain/avatar-group/avatar-group.tsx (hunk 2 above)
Comment (NidhiKJha) [reply ui/components/multichain/avatar-group/avatar-group.tsx]: Updated

File: ui/components/multichain/avatar-group/avatar-group.tsx (hunk 1 above)
Comment (NidhiKJha) [reply ui/components/multichain/avatar-group/avatar-group.tsx]: Updated here https://github.com/MetaMask/metamask-extension/pull/21342/commits/ce1a01ffa4aca65c367de0a7c1b4360854530e77

**Review by darkwing — APPROVED** [review]

**Review by vthomas13 — APPROVED** [review]

**Pull request merged**